radlab-ui/webapp/src/components/forms/StepCreator.tsx

```diff
@@ -59,24 +60,35 @@ const StepCreator: React.FC<StepCreator> = ({ variableList, idx }) => {
     return error
   }
   const renderControls = (variable: IUIVariable) => {
-    switch (variable.type) {
-      case "string":
-        return <StringField variable={variable} validate={validate(variable)} />
-      case "bool":
-        return <BooleanField variable={variable} />
-      case "number":
-        return <NumberField variable={variable} validate={validate(variable)} />
-      case "set(string)":
-      case "set(number)":
-        return <SetField variable={variable} />
-      case "list(string)":
-      case "list(number)":
-        return <ListField variable={variable} />
-      case "map":
-      case "map(string)":
-        return <MapField variable={variable} validate={validate(variable)} />
-      default:
-        return <StringField variable={variable} validate={validateRequired} />
+    let checkMapObject = variable.type.slice(0, 10)
+    if (checkMapObject === "map(object") {
```

Comment:
This looks wrong `"map(object"`

Reply:
how to check different scenario mapobject type like this
![image](https://user-images.githubusercontent.com/107676111/213443381-78febbec-7180-470d-a30c-88c5ce6e1242.png)

Reply:
updated with "mapobject"

Reply:
Now that I understand _why_ you had it as it was, I like the way it was previously. Can you revert this change?